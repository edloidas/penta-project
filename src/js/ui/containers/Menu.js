// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import MainMenu from '../components/MainMenu/MainMenu';
import { type MainMenuItemProps } from '../components/MainMenu/MainMenuItem';
import * as GameActions from '../actions/game';

type Props = {
  actions: {
    resumeGame: typeof GameActions.resumeGame
  }
};

// eslint-disable-next-line react/prefer-stateless-function
class Menu extends Component<Props> {
  constructor(props) {
    super(props);
    this.handleResume = this.handleResume.bind(this);
  }

  props: Props;

  handleResume: (e?: MouseEvent) => void;

  handleResume(e: MouseEvent) {
    e.stopPropagation();
    e.preventDefault();
    this.props.actions.resumeGame(true);
  }

  render() {
    const menuItems: Array<MainMenuItemProps> = [
      { name: 'Resume', clickHandler: this.handleResume },
      { name: 'New Game', to: '/new' },
      { name: 'Save Game', to: '/save' },
      { name: 'Load Game', to: '/load' },
      { name: 'Hero', to: '/hero' },
      { name: 'Settings', to: '/settings' },
      { name: 'Exit', to: '/exit' }
    ];
    return <MainMenu menuItems={menuItems} />;
  }
}

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch: Function) {
  return {
    actions: bindActionCreators(GameActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
