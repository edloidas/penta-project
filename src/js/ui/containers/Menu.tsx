import * as React from 'react';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import MainMenu from '../components/MainMenu/MainMenu';
import { MainMenuItemProps } from '../components/MainMenu/MainMenuItem';
import * as GameActions from '../actions/game';
import Focusable from './Focusable';

type Props = {
  actions: {
    resumeGame: typeof GameActions.resumeGame,
    startNewGame: typeof GameActions.startNewGame,
    saveGame: typeof GameActions.saveGame,
    loadGame: typeof GameActions.loadGame
  }
};

// eslint-disable-next-line react/prefer-stateless-function
class Menu extends Focusable<Props> {
  constructor(props) {
    super(props);
    this.handleResume = this.handleResume.bind(this);

    this.menuItems = [
      { name: 'Resume', clickHandler: this.handleResume },
      { name: 'New Game', to: '/new' },
      { name: 'Save Game', to: '/save' },
      { name: 'Load Game', to: '/load' },
      { name: 'Hero', to: '/hero' },
      { name: 'Settings', to: '/settings' },
      { name: 'Exit', to: '/exit' }
    ];
  }

  props: Props;

  menuItems: Array<MainMenuItemProps>;

  handleResume(e: MouseKeyboardEvent) {
    e.stopPropagation();
    e.preventDefault();
    this.props.actions.resumeGame(true);
  }

  render() {
    return (
      <div
        tabIndex={0}
        role="button"
        ref={div => {
          this.focusable = div;
        }}>
        <MainMenu menuItems={this.menuItems} />
      </div>
    );
  }
}

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch: Dispatch<GameActions.ActionsTypes>) {
  return {
    actions: bindActionCreators(GameActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
