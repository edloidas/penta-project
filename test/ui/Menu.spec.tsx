import * as React from 'react';
import { mount } from '../wrap';
import Menu from '../../src/js/ui/containers/Menu';

describe('<Menu />', () => {
  test('Should renders', () => {
    const tree = mount(<Menu />);
    expect(tree).toMatchSnapshot();
  });
});
