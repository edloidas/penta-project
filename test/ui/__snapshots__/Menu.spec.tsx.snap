// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should renders 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Menu)>
        <Menu
          actions={
            Object {
              "loadGame": [Function],
              "resumeGame": [Function],
              "saveGame": [Function],
              "startNewGame": [Function],
            }
          }
        >
          <div
            role="button"
            tabIndex={0}
          >
            <MainMenu
              menuItems={
                Array [
                  Object {
                    "clickHandler": [Function],
                    "name": "Resume",
                  },
                  Object {
                    "name": "New Game",
                    "to": "/new",
                  },
                  Object {
                    "name": "Save Game",
                    "to": "/save",
                  },
                  Object {
                    "name": "Load Game",
                    "to": "/load",
                  },
                  Object {
                    "name": "Hero",
                    "to": "/hero",
                  },
                  Object {
                    "name": "Settings",
                    "to": "/settings",
                  },
                  Object {
                    "name": "Exit",
                    "to": "/exit",
                  },
                ]
              }
            >
              <styled.div
                className="effect__appear"
              >
                <div
                  className="effect__appear sc-bwzfXH iVThND"
                >
                  <styled.div>
                    <div
                      className="sc-gZMcBi qaFpZ"
                    >
                      <styled.div>
                        <div
                          className="sc-gqjmRU cBrrLK"
                        >
                          <Logo
                            renderAuthor={false}
                            theme={
                              Object {
                                "fontSize": 0.5,
                              }
                            }
                          >
                            <styled.div
                              className="font__mono"
                              theme={
                                Object {
                                  "fontSize": 0.5,
                                }
                              }
                            >
                              <div
                                className="font__mono sc-htpNat easqzJ"
                              >
                                <styled.div>
                                  <div
                                    className="sc-ifAKCX RfDSL"
                                  >
                                    Penta Project
                                  </div>
                                </styled.div>
                                <styled.div>
                                  <div
                                    className="sc-EHOje bMrtUk"
                                  >
                                    A Cyberpunk Action Game
                                  </div>
                                </styled.div>
                              </div>
                            </styled.div>
                          </Logo>
                        </div>
                      </styled.div>
                      <MainMenuItem
                        clickHandler={[Function]}
                        key="Resume"
                        name="Resume"
                      >
                        <Button
                          clickHandler={[Function]}
                          name="Resume"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Resume"
                            onClick={[Function]}
                            tabIndex={0}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Resume"
                              onClick={[Function]}
                              tabIndex={0}
                            >
                              Resume
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="New Game"
                        name="New Game"
                        to="/new"
                      >
                        <Button
                          name="New Game"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/new"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="New Game"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="New Game"
                              tabIndex={-1}
                            >
                              New Game
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/new"
                              >
                                <a
                                  href="/new"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="Save Game"
                        name="Save Game"
                        to="/save"
                      >
                        <Button
                          name="Save Game"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/save"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Save Game"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Save Game"
                              tabIndex={-1}
                            >
                              Save Game
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/save"
                              >
                                <a
                                  href="/save"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="Load Game"
                        name="Load Game"
                        to="/load"
                      >
                        <Button
                          name="Load Game"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/load"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Load Game"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Load Game"
                              tabIndex={-1}
                            >
                              Load Game
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/load"
                              >
                                <a
                                  href="/load"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="Hero"
                        name="Hero"
                        to="/hero"
                      >
                        <Button
                          name="Hero"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/hero"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Hero"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Hero"
                              tabIndex={-1}
                            >
                              Hero
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/hero"
                              >
                                <a
                                  href="/hero"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="Settings"
                        name="Settings"
                        to="/settings"
                      >
                        <Button
                          name="Settings"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/settings"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Settings"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Settings"
                              tabIndex={-1}
                            >
                              Settings
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/settings"
                              >
                                <a
                                  href="/settings"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                      <MainMenuItem
                        key="Exit"
                        name="Exit"
                        to="/exit"
                      >
                        <Button
                          name="Exit"
                          theme={
                            Object {
                              "direction": "column",
                              "textAlign": "left",
                              "width": 19,
                            }
                          }
                          to="/exit"
                        >
                          <styled.button
                            className="effect__glitch"
                            data-text="Exit"
                            tabIndex={-1}
                            theme={
                              Object {
                                "direction": "column",
                                "textAlign": "left",
                                "width": 19,
                              }
                            }
                          >
                            <button
                              className="effect__glitch sc-iwsKbI ixchWs"
                              data-text="Exit"
                              tabIndex={-1}
                            >
                              Exit
                              <Link
                                replace={false}
                                tabIndex={0}
                                to="/exit"
                              >
                                <a
                                  href="/exit"
                                  onClick={[Function]}
                                  tabIndex={0}
                                />
                              </Link>
                            </button>
                          </styled.button>
                        </Button>
                      </MainMenuItem>
                    </div>
                  </styled.div>
                  <styled.footer
                    className="font__accent"
                  >
                    <footer
                      className="font__accent sc-VigVT byUSmW"
                    >
                      <styled.div>
                        <div
                          className="sc-jTzLTM iHsIJR"
                        >
                          <Version
                            theme={
                              Object {
                                "fontSize": 1,
                              }
                            }
                          >
                            <styled.span
                              theme={
                                Object {
                                  "fontSize": 1,
                                }
                              }
                            >
                              <span
                                className="sc-bZQynM fjiOsu"
                              >
                                <styled.span>
                                  <span
                                    className="sc-gzVnrw hjuePL"
                                  >
                                    alpha version
                                  </span>
                                </styled.span>
                                 
                                "0.0.1-alpha.2"
                              </span>
                            </styled.span>
                          </Version>
                        </div>
                      </styled.div>
                      <Copyright
                        theme={
                          Object {
                            "fontSize": 1,
                          }
                        }
                      >
                        <styled.span
                          theme={
                            Object {
                              "fontSize": 1,
                            }
                          }
                        >
                          <span
                            className="sc-htoDjs dNFyTW"
                          >
                            2017–current © 
                            edloidas production
                            . 
                            <styled.span>
                              <span
                                className="sc-dnqmqq oqlux"
                              >
                                All Rights Reserved.
                              </span>
                            </styled.span>
                          </span>
                        </styled.span>
                      </Copyright>
                    </footer>
                  </styled.footer>
                </div>
              </styled.div>
            </MainMenu>
          </div>
        </Menu>
      </Connect(Menu)>
    </Router>
  </ConnectedRouter>
</Provider>
`;
